electrumsv/tests/test_wallet.py

```diff
@@ -486,128 +508,346 @@ def test_multisig(self, mock_app_state, tmp_storage) -> None:
+        - derivation_data in the `MasterKeys` table contains various types of encrypted data
+          depending on the type of wallet or how old it is. These are the possible lookup keys in the
+          json:
+            - "seed"
+            - "passphrase"
+            - "xprv"
+        - encrypted_api_key data in the `Servers` table
+
+    New encrypted data may be added in future versions. Therefore, it is important to write this
+    test in such a way that it:
+        a) Captures any regressions in the re-encryption of migrated old wallets
+        b) Is easy to extend to cover any new encrypted data that is added in future versions
 
-    temp_dir = tempfile.mkdtemp()
-    for _wallet_filename in wallet_filenames:
-        source_wallet_path = os.path.join(TEST_WALLET_PATH, _wallet_filename)
-        wallet_path = os.path.join(temp_dir, _wallet_filename)
-        shutil.copyfile(source_wallet_path, wallet_path)
+    """
+    total_wallets_tested: int
+    failing_wallets: set[str]
+
+    # ---- Encrypted Data Types Requiring Test Coverage For Re-encryption ---- #
+
+    # Password token is re-encrypted in the `WalletData` table (included for a consistent approach
+    # of tracking coverate of each of the aforementioned categories of encrypted data)
+    count_coverage_password_token: int
+
+    # Derivation data in the KeyInstances table.
+    count_coverage_privkeys_in_keyinstance_table: int  # only for ImportedPrivkeyAccount types
+
+    # Derivation data in the MasterKeys table.
+    count_coverage_masterkeys_table_seed: int
+    count_coverage_masterkeys_table_passphrase: int
+    count_coverage_masterkeys_table_xprv: int
+
+    # Newer wallets have a `Servers` table with an `encrypted_api_key` column
+    count_coverage_server_api_key: int
+
+    # Note that while the CredentialCache does contain sensitive data, it is not encrypted and
+    # therefore does not need to be checked for re-encryption coverage.
+
+
+class TestLegacyWalletLoading:
+
+    re_encryption_test_state = ReEncryptionTestState(
+        total_wallets_tested=0,
+        failing_wallets=set(),
+        count_coverage_password_token=0,
+        count_coverage_privkeys_in_keyinstance_table=0,
+        count_coverage_masterkeys_table_seed=0,
+        count_coverage_masterkeys_table_passphrase=0,
+        count_coverage_masterkeys_table_xprv=0,
+        count_coverage_server_api_key=0
+    )
+    @pytest.mark.parametrize("storage_info,total_expected_wallets",
+        get_categorised_files2(TEST_WALLET_PATH, exclude_suffix=".json"))
+    @unittest.mock.patch('electrumsv.keystore.app_state.credentials.remove_indefinite_credential',
+        lambda x: None)
+    @unittest.mock.patch('electrumsv.wallet.app_state', new_callable=_create_mock_app_state)
+    def test_legacy_wallet_loading(self, mock_wallet_app_state, storage_info: WalletStorageInfo,
+            total_expected_wallets, set_to_mainnet_network_on_test_finish, caplog) -> None:
```

Comment:
caplog what even is this?
again set_to_mainnet_network_on_test_finish should be a decorator.

Reply:
I wondered the same thing about `caplog` but it was there before so I left it in...

Reply:
What do you mean by `set_to_mainnet_network_on_test_finish` should be a decorator? It's doing its job how it is. I am not familiar with pytest fixtures being used a decorators...